ckan/templates/snippets/search_result_text.html

```diff
@@ -28,18 +28,25 @@
   {% set text_query_none = _('No organizations found for "{query}"') %}
   {% set text_no_query = ungettext('{number} organization found', '{number} organizations found', count) %}
   {% set text_no_query_none = _('No organizations found') %}
+
+{% else %}
+  {% set text_query = ungettext('{number} {type} found for "{query}"', '{number} {type}s found for "{query}"', count) %}
```

Comment:
let's substitute the type before passing the string to ungettext. That lets extensions define their own plurals and translations of the types. (the list of types will be static for a given site so static translations are fine)